sql/catalyst/src/main/scala/org/apache/spark/sql/types/AbstractDataType.scala

```diff
@@ -96,6 +96,8 @@ private[sql] class TypeCollection(private val types: Seq[AbstractDataType])
 
 private[sql] object TypeCollection {
 
+  val Ordered = TypeCollection(NumericType, StringType)
```

Comment:
we have `abstract class AtomicType`, why not add an object for it like we did for numeric?